lib/spider/scripts/in_context_learning.py

```diff
@@ -1,50 +1,54 @@
-"""To run the script, use the following command:
-python spider/in_context_learning.py -m prompts=default,icl-1,icl-2,icl-3,icl-4  \
-pretrained_model_name_or_path=mistralai/Mixtral-8x7B-Instruct-v0.1
+"""Script to run an in-context-learning experiment.
 
-It will run the sweep over the different prompts and the pretrained model.
+To run the script, here is an example command command:
+    python scripts/in_context_learning.py -m \
+        prompts=default-1,default-2,icl-1,icl-2,icl-3,icl-4 \
+        pretrained_model_name_or_path=mistralai/Mixtral-8x7B-Instruct-v0.1
+
+It will run the sweep over the different prompts and parameters provided.
 """
 
 import json
 import os
 
 import clearml
 import hydra
-import nltk
 import torch
 import transformers
 import wandb
 from clearml import Task
 from omegaconf import OmegaConf
 from tqdm import tqdm
-from transformers import AutoModelForCausalLM, AutoTokenizer
+from transformers import AutoModelForCausalLM, AutoTokenizer, BitsAndBytesConfig
+from wandb import wandb_run
 
-from spider._path import _ROOT, DATASET_PATH
+from spider._path import CONFIG_PATH, DATASET_PATH, ENV_FILE_PATH
 from spider.clearml_utils import setup_clearml
 from spider.evaluation import build_foreign_key_map_from_json, evaluate
 from spider.utils import extract_sql, is_valid_sql
 
-nltk.download("punkt")
-
-
 OUTPUT_FILE = DATASET_PATH / "output.txt"
 gold = DATASET_PATH / "dev_gold.sql"
 db_dir = DATASET_PATH / "database"
 table = DATASET_PATH / "tables.json"
 
 
-@hydra.main(version_base="1.3", config_path=_ROOT / "configs", config_name="eval.yaml")
+@hydra.main(version_base="1.3", config_path=str(CONFIG_PATH), config_name="eval.yaml")
 def main(cfg) -> None:
     transformers.set_seed(cfg.seed)
+
     # loggers
-    setup_clearml(env_file_path=_ROOT / ".env")
+    setup_clearml(env_file_path=ENV_FILE_PATH)
     task = Task.init(
         project_name="spider",
         task_name="vanilla_llm",
         task_type=clearml.Task.TaskTypes.testing,
     )
     run = wandb.init(project="spider")
+    if not isinstance(run, wandb_run.Run):
+        raise TypeError("Run is not a valid Wandb run")
```

Comment:
It's a bit nitty, but it fixed a complaint from my LSP (having 0 complaint helps me to keep track of the issues I introduce)

The issue was that `wandb.init` can output a `wandb_run.Run`, `None` or something else, so the type checker can't know which one is used, so it complains when you call `run.some_method` because `None` doesn't support this method

Reply:
(this simply results in a sanity check, this should never be raised, except if I missed it up!)